Upgrade support for 1.10.0-beta.2

skinny opened the pull request:
The upgrade scenario was not working (yet). These changes enabled me to upgrade my 1.9.3. cluster to 1.10.0-beta.2

```
NAME                     STATUS    ROLES     AGE       VERSION
k8s-default-39042091-0   Ready     agent     22h       v1.9.3
k8s-master-39042091-0    Ready     master    22h       v1.10.0-beta.2

```

```

NAME                     STATUS    ROLES     AGE       VERSION
k8s-default-39042091-1   Ready     agent     5m        v1.10.0-beta.2
k8s-master-39042091-0    Ready     master    22h       v1.10.0-beta.2

```

Listing available upgrades for 1.10.0-beta.2 :

```{
  "orchestrators": [
    {
      "orchestratorType": "Kubernetes",
      "orchestratorVersion": "1.10.0-beta.2",
      "upgrades": [
        {
          "orchestratorType": "",
          "orchestratorVersion": "1.10.0-beta.3"
        }
      ]
    }
  ]
}
```

/cc @jackfrancis

jackfrancis commented:
Thanks @skinny! You'll need to remove the test starting at this line in order to get unit tests to pass:

https://github.com/Azure/acs-engine/blob/master/pkg/api/orchestrators_test.go#L93

Thanks for doing this testing/validation! :)

skinny commented:
@jackfrancis Will try to fix that later today. After a successful upgrade to beta2 I tried to do this same trick with beta3 but the engine doesn't see the beta3 as an upgrade to beta2. Any  suggestions on how (where) to fix this ?

skinny commented:
Tidied up the versions and added 1.9.4 and 1.10.0-beta3 for upgrade scenarios. I changed a couple of tests to accomodate for the new versions and they now all pass.

CecileRobertMichon commented on pkg/api/common/helper_test.go.orig:
```diff
@@ -0,0 +1,139 @@
+		t.Errorf("It is not Kubernetes version %s", KubernetesVersion1Dot6Dot13)
+	}
+
+<<<<<<< HEAD
```
you have a few unresolved merge conflicts in this file

skinny commented:
1.9.4 test still fail because it's not clear to me if Windows is supported or not atm. 

```validate_test.go:628: should not error on valid Windows version: Orchestrator Kubernetes version 1.9.4 does not support Windows```

skinny commented:
Removed the beta3 bits for now. Will create a new PR after this one is merged together with the other changes made today
